server.cfg

```diff
@@ -0,0 +1,22 @@
+# Servers are listed in the order they should be  registered.
+#
+
+# How to rebuild with updated server configurations: Modify the list below and
+# run `go generate && go build`
+
+# The parser takes the input format of:
+#
+#     <transport scheme>:<transport port>:<server creator function>
+# Or
+#     <plugin-name>:<fully-qualified-package-name>.<server creator function>
+#
+# External server example:
+#
+# quic:4242:github.com/coredns/coredns/dnsserver/experimental.NewServerQuic
+# Local plugin example:
+# tls:853:NewServerTLS
+
+tls:853:NewServerTLS
```

Comment:
If you instead have a convention that it is always `NewServer`, and instead put each different server creation routine in its own package, then you can generate the correct imports and calls to in `zservers.go` such that external servers become possible. As it is here, wouldn't you need new servers defined in the `dnsserver` package?

Reply:
so, for example

```
tls:853:tls
dns:53:dns
foo:10000:github.com/myrepo/fooserver
```

would become

```
import "github.com/coredns/coredns/core/dnsserver/tlsserver" tlssrvr
import "github.com/coredns/coredns/core/dnsserver/dnsserver" dnssrvr
import "github.com/myrepo/fooserver" foosrvr

...
...

	RegisterTransportServer(transport.TLS, transport.TLSPort, func(addr string, group []*Config) (caddy.Server, error) {
		return tlssrv.NewServer(addr, group)
	})

	RegisterTransportServer(transport.DNS, transport.DNSPort, func(addr string, group []*Config) (caddy.Server, error) {
		return dnssrv.NewServer(addr, group)
	})

	RegisterTransportServer(transport.FOO, transport.FOOPort, func(addr string, group []*Config) (caddy.Server, error) {
		return foosrv.NewServer(addr, group)
	})

...
....

Reply:
Yes, I thought about that but I was trying to minimize the changes. It would be cleaner to have an interface for it, do you think I should make that change?